src/Symfony/Component/Workflow/MarkingStore/MethodMarkingStore.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Workflow\MarkingStore;
+
+use Symfony\Component\Workflow\Marking;
+
+/**
+ * MethodMarkingStore stores the marking with a subject's method.
+ *
+ * This store deals with a "single state" or "multiple state" Marking.
+ *
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ */
+class MethodMarkingStore implements MarkingStoreInterface
+{
+    private $property;
+    private $singleState;
+
+    /**
+     * @param string $property Used to determine methods to call
+     *                         The `getMarking` method will use `$subject->getProperty()`
+     *                         The `setMarking` method will use `$subject->setProperty(string|array $places, array $context = array())`
+     */
+    public function __construct(string $property = 'marking', bool $singleState = false)
+    {
+        $this->property = $property;
+        $this->singleState = $singleState;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getMarking($subject)
+    {
+        $marking = $subject->{'get'.ucfirst($this->property)}();
```

Comment:
what if method does not exists?

Reply:
It will lead to a fatal error, and so an exception.

Reply:
yes obviously, but is this what you want no specific handling?

Reply:
We could. Indeed. I am thought it was enough but I will update the code

Reply:
I updated the code to throw an exception